Makefile

```diff
@@ -11,6 +11,39 @@ endif
 SHELL = /usr/bin/env bash -o pipefail
 .SHELLFLAGS = -ec
 
+PROJECT_DIR := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
+REPO := github.com/kubeflow/trainer
+TRAINER_CHART_DIR := $(PROJECT_DIR)/charts/kubeflow-trainer
+# Location to install tool binaries
+LOCALBIN ?= $(PROJECT_DIR)/bin
+
+# Tool versions
+K8S_VERSION ?= 1.32.0
+CONTROLLER_GEN_VERSION ?= v0.17.2
+GINKGO_VERSION ?= $(shell go list -m -f '{{.Version}}' github.com/onsi/ginkgo/v2)
+ENVTEST_VERSION ?= release-0.20
+ENVTEST_K8S_VERSION ?= 1.32
+HELM_VERSION ?= v3.15.3
+HELM_UNITTEST_VERSION ?= 0.5.1
+HELM_CHART_TESTING_VERSION ?= v3.12.0
+HELM_DOCS_VERSION ?= v1.14.2
+YQ_VERSION ?= v4.45.1
+
+# Tool binaries
+CONTROLLER_GEN ?= $(LOCALBIN)/controller-gen
+GINKGO ?= $(LOCALBIN)/ginkgo
+ENVTEST ?= $(LOCALBIN)/setup-envtest
+KIND ?= $(LOCALBIN)/kind
+HELM ?= $(LOCALBIN)/helm
+HELM_DOCS ?= $(LOCALBIN)/helm-docs
+YQ ?= $(LOCALBIN)/yq
+
+# CONTAINER_TOOL defines the container tool to be used for building images.
+# Be aware that the target commands are only tested with Docker which is
+# scaffolded by default. However, you might want to replace it to use other
+# tools. (i.e. podman)
+CONTAINER_TOOL ?= docker
```

Comment:
Can we remove the container tool for now and only rely on `docker` please ?
I think, we should discuss separately if we should allow users to use other container tools to build images (e.g. podman).
For now we use docker to build our images:
https://github.com/kubeflow/trainer/blob/master/hack/e2e-setup-cluster.sh#L37